controller/share_manager_controller.go

```diff
@@ -740,8 +742,42 @@ func (c *ShareManagerController) createShareManagerPod(sm *longhorn.ShareManager
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to create pod for share manager %v", sm.Name)
 	}
+
+	ret, err := c.verifyPodSpecNodeName(pod.Name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to verify creation of pod %v", pod.Name)
+	}
+
 	getLoggerForShareManager(c.logger, sm).WithField("pod", pod.Name).Info("Created pod for share manager")
-	return pod, nil
+	return ret, nil
+}
+
+func (c *ShareManagerController) verifyPodSpecNodeName(podName string) (*corev1.Pod, error) {
```

Comment:
Similarly, do we need to put this into `ds.CreatePod()`?

Reply:
Remove this commit and will be handled in https://github.com/longhorn/longhorn/issues/5233.
Let the PR focus on the configmap operations.